### ISSUE TITLE
Selector in mapState called with empty state

### ISSUE THREAD

**Issue by eric-burel** [rematch/rematch#386]
Sorry I can't provide much info about this, but selectors seems to be called once before the state is defined.

This might come from the connection pattern:
```js
const mapState = state => ({
  ...state.perimeter,
  loadingDatabases: state.loading.effects.perimeter.getDatabases,
  loadingSchemas: state.loading.effects.perimeter.getSchemas,
  hasSelectedAtLeastOneSchema: select.perimeter.hasSelectedAtLeastOneSchema(
    state
  )
});
```
The `mapState` seems to be called once before the state is fully initialized, so it make the selector fail. Quickfix is to check if the state is defined, but this is not great.
Maybe I misuse selectors?

**Comment by d3dc** [comment]
@eric-burel `select@2.0.0` is pretty different. Any changes?

Your selector gets called when your component renders. If you are doing `rematch.init()` before `react.render()`, you should be safe. Do you have asynchronous state?

**Comment by eric-burel** [comment]
Hi, I did not test with the newer version as I don't maintain the related app. You can close if this should be fixed un 2.0.
